contracts/cw1-subkeys/src/error.rs

```diff
@@ -0,0 +1,47 @@
+use cosmwasm_std::StdError;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ContractError {
```

Comment:
Nice explicit error cases. However, I didn't see them all checked in the test cases. I think there are some which would be even cleaner when they assert eg. `ContractError::DelegatePerm` to see exactly what failed.

Reply:
That's true. Let's add extra checks in a different task / PR.